src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,101 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface NewMovieProps {
+  onAdd: React.Dispatch<React.SetStateAction<Movie[]>>;
+}
+
+export const NewMovie: React.FC<NewMovieProps> = ({ onAdd }) => {
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [count, setCount] = useState(0);
+  const [disabled, setDisabled] = useState(true);
+
+  const resetForm = () => {
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  useEffect(() => {
+    if (title && imdbId && imdbUrl && imgUrl) {
+      setDisabled(false);
+    }
+  }, [title, imdbUrl, imgUrl, imdbId]);
```

Comment:
it is redundant to make state for such codition. enough to make variable. and looks like it is should be with || logical operator, as button should be disabled when some required input is empty.
trim should solve problem in screencast (now possible add movie with only spaces in input)

```suggestion
 const isDisabled = title.trim() ||  imdbId.trim() || imdbUrl.trim() || imgUrl.trim()
```
![chrome-capture-2024-1-6](https://github.com/mate-academy/react_movies-list-add-form/assets/33029864/93dd52a0-d3d4-47af-9733-da84d2cf0c86)
 
![Screenshot from 2024-01-06 17-41-55](https://github.com/mate-academy/react_movies-list-add-form/assets/33029864/c8e5b2fd-32a5-4981-8d85-bea719169daf)